src/GraphQL/Types/Scalars/SByteGraphType.cs

```diff
@@ -12,25 +12,56 @@ public class SByteGraphType : ScalarGraphType
         /// <inheritdoc/>
         public override object ParseLiteral(IValue value) => value switch
         {
-            IntValue intValue => sbyte.MinValue <= intValue.Value && intValue.Value <= sbyte.MaxValue ? (sbyte?)intValue.Value : null,
-            LongValue longValue => sbyte.MinValue <= longValue.Value && longValue.Value <= sbyte.MaxValue ? (sbyte?)longValue.Value : null,
-            BigIntValue bigIntValue => sbyte.MinValue <= bigIntValue.Value && bigIntValue.Value <= sbyte.MaxValue ? (sbyte?)bigIntValue.Value : null,
-            _ => null
+            IntValue intValue => checked((sbyte)intValue.Value),
+            LongValue longValue => checked((sbyte)longValue.Value),
+            BigIntValue bigIntValue => checked((sbyte)bigIntValue.Value),
+            NullValue _ => null,
+            _ => ThrowLiteralConversionError(value)
         };
 
-        /// <inheritdoc/>
-        public override object ParseValue(object value) => ValueConverter.ConvertTo(value, typeof(sbyte));
-
         /// <inheritdoc/>
         public override bool CanParseLiteral(IValue value) => value switch
         {
             IntValue intValue => sbyte.MinValue <= intValue.Value && intValue.Value <= sbyte.MaxValue,
             LongValue longValue => sbyte.MinValue <= longValue.Value && longValue.Value <= sbyte.MaxValue,
             BigIntValue bigIntValue => sbyte.MinValue <= bigIntValue.Value && bigIntValue.Value <= sbyte.MaxValue,
+            NullValue _ => true,
             _ => false
         };
 
         /// <inheritdoc/>
-        public override IValue ToAST(object value) => new IntValue(Convert.ToSByte(value));
+        public override object ParseValue(object value) => value switch
+        {
+            sbyte _ => value,
+            null => null,
+            int i => checked((sbyte)i),
+            byte b => checked((sbyte)b),
+            short s => checked((sbyte)s),
+            ushort us => checked((sbyte)us),
+            uint ui => checked((sbyte)ui),
+            long l => checked((sbyte)l),
+            ulong ul => checked((sbyte)ul),
+            BigInteger bi => (sbyte)bi,
+            float f => checked((sbyte)f),
```

Comment:
Unfortunately such work with floating point numbers results in loss of accuracy. See `SByteGraphTypeTests.Coerces_given_inputs_to_overflow_exception`, case `999.99`. Add `1.3` case. It fails since `checked((sbyte)1.3)` yields `1`. That is why when you asked whether it was necessary to add new cases for all the numeric scalars for parsing, then I said yes but maybe exclude floating-point ones exactly because truncation problem. The same problem was in past with `DateGraphType` that "normally" worked with time component simply truncating it. Now it throw exception `Expected date to have no time component` to prevent loss of accuracy. 

You may add something like
```csharp
        private static sbyte Ret(float f) => checked((sbyte)f == f ? (sbyte)f : throw new InvalidOperationException("Loss of accuracy for scalar 'XXX' parsing 'value'."));
```

Reply:
I wouldn't.  We support floating-point values because of how the serializers work.  They are only supported in ParseValue, not ParseLiteral.  And by definition, `(sbyte)f == f` may return false because 3.0 is not stored as exactly 3.  And I don't think we should be writing `Math.Abs(f - (float)(sbyte)f) < float.Epsilon` which is the proper way to perform this check. If we want to "fix" the problem, we should build merge the deserialization engine with GraphQL, so that it can read the json data in the correct format the first time, rather than having a bunch of TryGet code (see ObjectDictionaryConverter) and then have to coerce the value.

Reply:
> See SByteGraphTypeTests.Coerces_given_inputs_to_overflow_exception, case 999.99. Add 1.3 case. It fails since checked((sbyte)1.3) yields 1

I am interested in what to do now in the case of 1.3 for `sbyte`.

Reply:
I do not urge to use math. You can generally abandon the support of the convertion from floating point numbers. I'm talking about that now the code works incorrectly for the case described above.

Reply:
> I am interested in what to do now in the case of 1.3 for `sbyte`.

We can introduce proper math, as described above, to check for this scenario.  We cannot add `(sbyte)f == f`.  I would not do this.  Or we can ignore the error, just as it was done before.

Note that this is only for ParseValue, and only there because of the nature of json deserializers.  This is not a bug that needs to be fixed.

Reply:
>  Adding the check you suggested will lead to bugs for certain values on certain platforms.

It was just an example.

> If you like, perhaps we can remove the floating point conversions for the types that are int or smaller, hoping and assuming that the json deserializer uses an int data type rather than double during its deserialization process.

I'm OK with remove floating point conversions from both `ParseValue` and `Serialize`. It seems to me that integers and floating point numbers do not need to mix. We already have `IntValue` and `FloatValue` different literals for this reason.

> and only there because of the nature of json deserializers. This is not a bug that needs to be fixed.

I don't understand and disagree.

Reply:
I'll just write some tests to demonstrate the problem.  If the problem doesn't exist, then we don't need to worry about it.

Reply:
👍

Reply:
I was unable to reproduce the problem.  Looks like the json converters do not convert to floating point for integer numbers.  I'll remove the floating-point support from ParseValue.  I added tests for both json converters to confirm this.

Reply:
Add test for
```json
{
  "someBigNumber": 1646176147698146586816969619696946176496969696996969869
}
```

After all, the deserialization of variables occurs in `Dictionary<string, object>`, right?

Reply:
Tests proved that the `ObjectDictionaryConverter` needed to be changed to give priority to `BigInteger` over `double` and `decimal` or else a large value that can be contained "without loss of precision" as a floating-point value would not be stored as a `BigInteger`, breaking deserialization.  I've made the change.  No issue with newtonsoft.

Reply:
Hmm, I did not think about it when I added `BigInteger` support. That is, the change should lead to the fact that we will become in more cases to deserialize the number into `BigInteger` instead of `double`?

> No issue with newtonsoft.

Why?

Reply:
And we don't even have proper serialization support for BigInteger values.  I'm fixing that also.

Reply:
Removed floating-point parsing as requested.  Fixed BigInteger parsing and serialization (both broken).

Reply:
> Hmm, I did not think about it when I added `BigInteger` support. That is, the change should lead to the fact that we will become in more cases to deserialize the number into `BigInteger` instead of `double`?
> 
> > No issue with newtonsoft.
> 
> Why?

Newtonsoft has built-in support for BigInteger that takes precedence over floating-point types.

Reply:
> Hmm, I did not think about it when I added `BigInteger` support. That is, the change should lead to the fact that we will become in more cases to deserialize the number into `BigInteger` instead of `double`?

Yes but the scalar will convert it to the proper type.  It will never be presented as a BigInteger, even if they access `Variables` directly, as they will be accessing the copy of the variables after it has been deserialized by the scalars via `ParseValue`.  It also should not be any slower, because already the code attempts to read it both as a `BigInteger` and a `double`.

Reply:
> Yes but the scalar will convert it to the proper type.

I understand.